tests/std/tests/P0288R9_move_only_function/test.cpp

```diff
@@ -0,0 +1,565 @@
+    f_c f3(test_const_t<true>{});
+    assert(f3() == 456);
+    const f_c f4(test_const_t<true>{});
+    assert(f4() == 456);
+}
+
+
+void test_qual() {
+    move_only_function<int(int)> f1([](auto i) { return i + 1; });
+    assert(f1(1) == 2);
+    move_only_function<int(int)&> f2([](auto i) { return i + 1; });
+    assert(f2(2) == 3);
+    move_only_function<int(int) &&> f3([](auto i) { return i + 1; });
+    assert(move(f3)(3) == 4);
+
+    move_only_function<int(int) const> f1c([](auto i) { return i + 1; });
+    assert(f1c(4) == 5);
+    move_only_function<int(int) const&> f2c([](auto i) { return i + 1; });
+    assert(f2c(5) == 6);
+    move_only_function<int(int) const&&> f3c([](auto i) { return i + 1; });
+    assert(move(f3c)(6) == 7);
+
+    move_only_function<int(int) noexcept> f1_nx([](auto i) noexcept { return i + 1; });
+    assert(f1_nx(1) == 2);
+    move_only_function<int(int)& noexcept> f2_nx([](auto i) noexcept { return i + 1; });
+    assert(f2_nx(2) == 3);
+    move_only_function<int(int)&& noexcept> f3_nx([](auto i) noexcept { return i + 1; });
+    assert(move(f3_nx)(3) == 4);
+
+    move_only_function<int(int) const noexcept> f1c_nx([](auto i) noexcept { return i + 1; });
+    assert(f1c_nx(4) == 5);
+    move_only_function<int(int) const& noexcept> f2c_nx([](auto i) noexcept { return i + 1; });
+    assert(f2c_nx(5) == 6);
+    move_only_function<int(int) const&& noexcept> f3c_nx([](auto i) noexcept { return i + 1; });
+    assert(move(f3c_nx)(6) == 7);
+}
+
+static_assert(is_same_v<move_only_function<void()>::result_type, void>);
+static_assert(is_same_v<move_only_function<short(long&) &>::result_type, short>);
+static_assert(is_same_v<move_only_function<int(char*) &&>::result_type, int>);
+static_assert(is_same_v<move_only_function<void() const>::result_type, void>);
+static_assert(is_same_v<move_only_function<short(long&) const&>::result_type, short>);
+static_assert(is_same_v<move_only_function<int(char*) const&&>::result_type, int>);
+
+#ifdef __cpp_noexcept_function_type
+static_assert(is_same_v<move_only_function<void() noexcept>::result_type, void>);
+static_assert(is_same_v<move_only_function<short(long&) & noexcept>::result_type, short>);
+static_assert(is_same_v<move_only_function<int(char*) && noexcept>::result_type, int>);
+static_assert(is_same_v<move_only_function<void() const noexcept>::result_type, void>);
+static_assert(is_same_v<move_only_function<short(long&) const& noexcept>::result_type, short>);
+static_assert(is_same_v<move_only_function<int(char*) const&& noexcept>::result_type, int>);
+#endif // ^^^ defined(__cpp_noexcept_function_type) ^^^
+
+bool fail_allocations = false;
+
+#pragma warning(suppress : 28251) // Inconsistent annotation for 'new': this instance has no annotations.
+void* operator new(size_t size) {
+    if (fail_allocations) {
+        throw bad_alloc{};
+    }
+    void* result = size ? malloc(size) : malloc(1);

```

Comment:
We conventionally avoid directly testing integers for being nonzero. (Test code conventions are somewhat weaker than product code, but this one is generally followed, I think).